### PR TITLE
Fix order of ComponentNames

### LINKED ISSUE #521: ComponentNames out of order
The HDF5 overhaul has led to the arrays `NumComponents` and `ComponentNames` in the output being out of order compared to `OutputDatasetNames`. 

I'm not sure how things should function with sparse variables in the output, but there should be enough information in `NumComponents`, `ComponentNames`, and `OutputDatasetNames` to build a dictionary of dictionaries such as `{"DatasetName":{"ComponentName":idx}}`

### PR DESCRIPTION
**Pull request by forrestglines**
## PR Summary

Fixes out of order `ComponentNames` and `NumComponents` introduced in  #486.

There should be enough information in `NumComponents`, `ComponentNames`, and `OutputDatasetNames` to build a dictionary of dictionaries such as `{"DatasetName":{"ComponentName":idx}}`.

Fixes #521 

## PR Checklist

- [x] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [ ] Changes are summarized in CHANGELOG.md
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [x] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Comment by forrestglines** [comment]
@jlippuner Could you take a look at this change? I discovered that the use of `std::set` led to `ComponentNames` being in the different order compared to `OutputDatasetNames`. I made a quick fix that works but it's not elegant. I'm also not sure if it's correct in the context of sparse variables.

Do you see a better solution to access the component labels with the variable metadata for each `std::string` in `output_params.variables` from this function? I'm struggling to see where to access them.

**Review by Yurlungur — APPROVED** [review]

**Comment by jlippuner** [comment]
> @jlippuner Could you take a look at this change? I discovered that the use of `std::set` led to `ComponentNames` being in the different order compared to `OutputDatasetNames`. I made a quick fix that works but it's not elegant. I'm also not sure if it's correct in the context of sparse variables.
> 
> Do you see a better solution to access the component labels with the variable metadata for each `std::string` in `output_params.variables` from this function? I'm struggling to see where to access them.

Sorry, only saw this now. Yes, there is indeed a mismatch between `OutputDatasetNames` and `ComponentNames`. However, I think we need to fix `OutputDatasetNames` rather than rearranging `ComponentNames`, because the `ComponentNames` is sorted alphabetically and it's the same order as the `SparseInfo` data.

I'll push what I think should fix the issue.

**Comment by jlippuner** [comment]
@forrestglines I think the order is now consistent. I also moved the check for empty component labels to the `VarInfo` creator.

**Review by jlippuner — APPROVED** [review]

File: src/outputs/parthenon_hdf5.cpp [hunk 1]
```diff
@@ -103,8 +104,10 @@ struct VarInfo {
 
   VarInfo(const std::string &label, const std::vector<std::string> &component_labels,
           int vlen, bool is_sparse, bool is_vector)
-      : label(label), component_labels(component_labels), vlen(vlen),
-        is_sparse(is_sparse), is_vector(is_vector) {
+      : label(label),
+        component_labels(component_labels.size() > 0 ? component_labels
+                                                     : std::vector<std::string>{label}),
```
Comment (JoshuaSBrown) [inline src/outputs/parthenon_hdf5.cpp]: Please use a different signature for component_labels the argument vs the member.

Comment (JoshuaSBrown) [reply src/outputs/parthenon_hdf5.cpp]: The same for the other arguments.

Comment (forrestglines) [reply src/outputs/parthenon_hdf5.cpp]: Yes - I renamed the members in `VarInfo` to have underscores

**Comment by forrestglines** [comment]
@par-hermes format

**Pull request merged**